psi4/src/psi4/libfock/jk.h

```diff
@@ -1193,94 +1182,134 @@ class PSI_API MemDFJK : public JK {
 };
 
 /**
- * Class DFJCOSK
+ * Class SplitJKBase
  *
- * JK implementation using a direct density-fitted coulomb algorithm
- * and a semi-numerical 'chain of spheres' exchange algorithm
+ * A wrapper class that allows building of different J and K algorithms
+ * (such as Direct-DF J builds, and LinK builds)
  */
-class PSI_API DFJCOSK : public JK {
-   protected:
-    /// Number of threads
-    int nthreads_;
-    /// Options object
-    Options& options_;
-    /// Previous iteration pseudo-density matrix
-    std::vector<SharedMatrix> D_prev_;
-
-    // => Density Fitting Stuff <= //
-
-    /// Auxiliary basis set
-    std::shared_ptr<BasisSet> auxiliary_;
-    /// Coulomb Metric
-    SharedMatrix J_metric_;
-    /// per-thread TwoBodyAOInt object (for computing three-center ERIs)
-    std::vector<std::shared_ptr<TwoBodyAOInt>> eri_computers_;
-
-    // => Semi-Numerical Stuff <= //
-
-    /// Small DFTGrid for initial SCF iterations
-    std::shared_ptr<DFTGrid> grid_init_;
-    /// Large DFTGrid for the final SCF iteration
-    std::shared_ptr<DFTGrid> grid_final_;
-    /// Overlap fitting metric for grid_initial_
-    SharedMatrix Q_init_;
-    /// Overlap fitting metric for grid_final_
-    SharedMatrix Q_final_;
-
-    std::string name() override { return "DFJCOSK"; }
-    size_t memory_estimate() override;
-
-    // => Required Algorithm-Specific Methods <= //
+class SplitJKBase {
```

Comment:
Open question to the reviewers: would it be best to have SplitJKBase defined in the jk.h file? I do wonder whether it would be better from a design perspective to include it in a separate header file, something like composite_helper.h

Reply:
@davpoolechem @loriab Very good point

Reply:
Made the change! SplitJKBase is now in composite.h